packages/extensions/venia-sample-backends/intercept.js

```diff
@@ -29,56 +35,61 @@ const fetchBackends = async debug => {
 };
 
 /**
- * Validation function to check if the backend being used is one of the sample backends provided
- * by PWA Studio. If yes, the function validates if the backend is active. If not, it reports an
- * error by calling the onFail function. In the error being reported, it sends the other sample
- * backends that the developers can use.
+ * A validation function that verifies that the backend used is active if it is
+ * one of the sample backends provided by Magento. If the backend is not active
+ * the function provides a list of alternatives.
  *
- * @param {Object} config.env - The ENV provided to the app, usually avaialable through process.ENV
+ * @param {Object} config.env - The ENV provided to the app, usually available through process.ENV
  * @param {Function} config.onFail - callback function to call on validation fail
  * @param {Function} config.debug - function to log debug messages in console in debug mode
  *
- * To watch the debug messages, run the command with DEBUG=*runEnvValidators*
+ * To watch the debug messages, run the command with DEBUG=pwa-buildpack:runEnvValidators
  */
 const validateSampleBackend = async config => {
-    console.warn(
-        "\n venia-sample-backends is a Dev only extension, please remove it from the project's package.json before going to production. \n"
-    );
-
     const { env, onFail, debug } = config;
 
-    const backendIsActive = await isBackendActive(env, debug);
+    const sampleBackends = await fetchBackends(debug);
+    const sampleBackendUrls = sampleBackends.map(({ url }) => url);
 
-    if (!backendIsActive) {
-        debug(`${env.MAGENTO_BACKEND_URL} is inactive`);
-
-        debug('Fetching other backends');
-
-        const sampleBackends = await fetchBackends(debug);
-        const otherBackends = sampleBackends.filter(
-            ({ url }) => url !== env.MAGENTO_BACKEND_URL
+    if (!sampleBackendUrls.includes(env.MAGENTO_BACKEND_URL)) {
+        // We don't need to validate anyone's custom backend. The validator is
+        // only for the sample backends we provide.
+        debug(`Not validating custom backend: ${env.MAGENTO_BACKEND_URL}`);
```

Comment:
This is important - if someone is playing around with PWA Studio using a sample backend, we want to inform them of alternatives if it goes down. But if someone has already swapped to their custom backend, we don't need to validate it.

Reply:
We dint do this on purpose. If we do this, irrespective of whether they are using a custom or sample backend, we will be hitting the Sample Backends URL to fetch the list, which will be a burden on the server. That's the reason why we only fetch the list if the given backend is down.

Reply:
Ah dang. Right. Hmm.